libs/common-system/src/main/java/org/codice/ddf/configuration/DictionaryMap.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A HashMap based Dictionary for use with OSGI methods requiring a Dictionary
+ *
+ * @param <K>
+ * @param <V>
+ */
+public class DictionaryMap<K, V> extends Dictionary<K, V> implements Map<K, V>, Serializable {
+  private HashMap<K, V> map;
+
+  public DictionaryMap(int initialCapacity, float loadFactor) {
+    map = new HashMap<>(initialCapacity, loadFactor);
+  }
+
+  public DictionaryMap(int initialCapacity) {
+    map = new HashMap<>(initialCapacity);
+  }
+
+  public DictionaryMap() {
+    map = new HashMap<>();
+  }
+
+  public DictionaryMap(Map<? extends K, ? extends V> m) {
+    map = new HashMap<>(m);
+  }
+
+  @Override
+  public int size() {
+    return map.size();
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return map.isEmpty();
+  }
+
+  @Override
+  public boolean containsKey(Object key) {
+    return map.containsKey(key);
+  }
+
+  @Override
+  public boolean containsValue(Object value) {
+    return map.containsValue(value);
+  }
+
+  @Override
+  public Enumeration<K> keys() {
+    return new EnumerationImpl<>(map.keySet().iterator());
+  }
+
+  @Override
+  public Enumeration<V> elements() {
+    return new EnumerationImpl<>(map.values().iterator());
```

Comment:
✏️  Can be replaced with org.apache.commons.collections4.iterators.IteratorEnumeration